src/main/java/org/wildfly/security/ssl/Protocol.java

```diff
@@ -29,40 +29,51 @@
     /**
      * The SSL version 2 protocol.
      */
-    SSLv2,
+    SSLv2 ("SSLv2"),
     /**
      * The SSL version 3 protocol.
      */
-    SSLv3,
+    SSLv3 ("SSLv3"),
     /**
      * The TLS version 1.0 protocol.
      */
-    TLSv1,
+    TLSv1 ("TLSv1"),
     /**
      * The TLS version 1.1 protocol.  Note that there are no cipher suites which are specifically defined in this protocol.
      */
-    TLSv1_1,
+    TLSv1_1 ("TLSv1.1"),
     /**
      * The TLS version 1.2 protocol.
      */
-    TLSv1_2,
+    TLSv1_2 ("TLSv1.2"),
     /**
      * The TLS version 1.3 protocol.
      */
-    TLSv1_3,
+    TLSv1_3 ("TLSv1.3"),
     ;
+
     static final int fullSize = values().length;
 
-    static Protocol forName(final String name) {
-        switch (name) {
-            case "SSLv2": return SSLv2;
-            case "SSLv3": return SSLv3;
-            case "TLSv1": return TLSv1;
-            case "TLSv1.1": return TLSv1_1;
-            case "TLSv1.2": return TLSv1_2;
-            case "TLSv1.3": return TLSv1_3;
-            default: return null;
+    public final String name;
+
+    Protocol(String name) {
+        this.name = name;
+    }
+
+    public static Protocol forName(final String name) {
+        for (Protocol protocol : values()) {
```

Comment:
In this case I think there should be a ``static final Map<String, Protocol> map`` which is initialized with the class.  Then this method can just do ``return map.get(name);`` instead of linearly scanning the list (though the list is small in this case).